README.md

```diff
@@ -98,6 +98,28 @@ files:
 
 When `output: none` is used, the `conda_dir`, `requirements_dir` and `matrix` keys can be omitted. The use case for `output: none` is described in the [_Additional CLI Notes_](#additional-cli-notes) section below.
 
+#### `extras`
+
+A given file may include an `extras` entry that may be used to provide inputs specific to a particular file type
+
+Here is an example:
+
+```yaml
+files:
+  build:
+    output: pyproject
+    includes: # a list of keys from the `dependencies` section which should be included in the generated files
+      - build
+    extras:
+      table: table_name
+      key: key_name
+```
+
+Currently the supported extras by file type are:
+- pyproject.toml
+  - table: The table in pyproject.toml where the dependencies should be written.
+  - key: The key corresponding to the dependency list in `table`.
```

Comment:
I understand why you chose to make it flexible like this, but I feel like this is not constrained enough since it allows us to generate invalid `pyproject.toml` files. It breaks my basic assumption for this tool that any valid `dependencies.yaml` file should _always_ generate valid output files.

I had suggested a constrained combination of `type: ['build', 'optional', 'standard'] = 'standard'` and `section: str = None` with noted defaults, because it still allows us to validate that `section` is only used in combination with `optional` and guarantees that we build a valid `pyproject.toml` file.

As another argument against this approach is that it requires us to _always_ specify the `extras` for this output type which is not only annoying, but also error-prone.

Reply:
Those are good points, I'll address them separately.
- I agree that we should constrain the options to avoid invalid choices. I think that is unquestionably an improvement. We should ensure that dependencies.yaml cannot produce an invalid output file.
- While I see the argument that always specifying the extras is somewhat error-prone, I'm not convinced that the tradeoff of inventing a new set of names is better. I would rather use the exact names that match, i.e. I prefer `table` to `type` and using the actual table name (build-system, project, project.optional-dependencies). Since effectively every project will end up specifying all three (definitely the first two) I don't think it makes sense to privilege one table over the others. We can still ensure that only a small set of expected names works.
- I would be fine making the key an optional field that can only be provided for optional dependencies since in the other cases it is fixed.

Reply:
I have the impression that we agree on the most important points.

Reply:
I believe that I've implemented what we want here.